generators/repo/src/commands/generate.ts

```diff
@@ -129,7 +130,7 @@ export class GenerateCommand implements RepomanCommand {
     }
 
     private updateRemote = async (repo: GitRepo, remote: GitRemote): Promise<RemotePushResult> => {
-        const defaultBranch = remote.branch || "main";
+        const defaultBranch = this.options.baseBranch || remote.branch || "main";
```

Comment:
If i recall, repoman supports having multiple remotes configured.  An optional implementation would be to configure multiple remotes to the same repo with different branches `main` vs `staging`.  Then we can use the `-r` flag to set the remote that should be the target for the generation.